clang/lib/AST/Stmt.cpp

```diff
@@ -990,6 +990,14 @@ bool IfStmt::isObjCAvailabilityCheck() const {
 }
 
 Optional<const Stmt*> IfStmt::getNondiscardedCase(const ASTContext &Ctx) const {
+  Optional<Stmt *> Result =
```

Comment:
FYI @AaronBallman : this is what this looks like if we implement in terms of the non-const version.  Unfortunately `Optional` doesn't have any better way of constructing it.  I would love it if `std::optional` (I wouldn't want `llvm::Optional` to diverge from the standard version) had a `Optional(Optional<const T>)` constructor... Wonder if that is worth spending time in LEWG for :) 

Let me know which you like better, I can revert this change if you hate it.

Reply:
I think this would be better expressed as:
```
if (Optional<Stmt *> Result = const_cast<IfStmt *>(this)->getNondiscardedCase(Ctx))
  return *Result;
return None;
```
which, tbh, isn't all that bad IMO. WDYT?